snapcraft/internal/states/_global_state.py

```diff
@@ -31,6 +34,24 @@ class GlobalState(State):
 
     yaml_tag = u"!GlobalState"
 
-    def __init__(self, build_packages, build_snaps):
+    @classmethod
+    def load(cls, *, filepath: str):
+        with open(filepath) as state_file:
+            return yaml.load(state_file)
+
+    def save(self, *, filepath: str) -> None:
+        dirpath = os.path.dirname(filepath)
+        if dirpath:
+            os.makedirs(dirpath, exist_ok=True)
+        with open(filepath, "w") as state_file:
+            yaml.dump(self, stream=state_file)
+
+    def get_build_packages(self) -> List[str]:
+        return self.assets.get("build-packages", [])
+
+    def get_build_snaps(self) -> List[str]:
+        return self.assets.get("build-snaps", [])
+
+    def __init__(self, *, build_packages: List[str], build_snaps: List[str]) -> None:
```

Comment:
Wait... this should totally be broken right now. [This](https://github.com/snapcore/snapcraft/pull/2186/files#diff-2e467cf166266ae6d2e2c42991c854f1R27) should be trying to pass `assets` as a kwarg, not `build_packages` or `build_snaps`. What's happening here?

Reply:
```
(snapcraft) ubuntu@snapcraft-xenial-dev:~/source/make-hello$ ipython
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from snapcraft.internal.states import GlobalState

In [2]: s = GlobalState.load(filepath="snap/.snapcraft/state")

In [3]: s
Out[3]: GlobalState(assets: OrderedDict([('build-packages', ['gcc']), ('build-snaps', ['go'])]))

In [4]: cat snap/.snapcraft/state
!GlobalState
assets:
  build-packages: [gcc]
  build-snaps: [go]
```